tests/suite-unit/PluginFlyvemdmTaskstatus.php

```diff
@@ -35,12 +35,48 @@
 
 class PluginFlyvemdmTaskstatus extends CommonTestCase {
 
+   public function beforeTestMethod($method) {
+      switch ($method) {
+         case 'testPrepareInputForAdd':
+         case 'testPrepareInputForUpdate':
+         case 'testDisplayTabContentForItem':
+         case 'testGetTabNameForItem':
+            $this->login('glpi', 'glpi');
+            break;
+      }
+   }
+
+   public function afterTestMethod($method) {
+      switch ($method) {
+         case 'testPrepareInputForAdd':
+         case 'testPrepareInputForUpdate':
+         case 'testDisplayTabContentForItem':
+         case 'testGetTabNameForItem':
+            parent::afterTestMethod($method);
+            \Session::destroy();
+            break;
+      }
+   }
+
    /**
-    * @return object
+    * @return array
     */
-   private function createInstance() {
-      $this->newTestedInstance();
-      return $this->testedInstance;
+   private function createFleetAndTask() {
+      $policy = new \PluginFlyvemdmPolicy();
+      $policy->getFromDbBySymbol('storageEncryption');
+      $fleet = $this->createFleet([
+         'name' => $this->getUniqueString(),
+      ]);
+      $task = new \PluginFlyvemdmTask();
+      $task->add([
+         'value'                       => '0',
+         'plugin_flyvemdm_policies_id' => $policy->getID(),
+         'itemtype_applied'            => \PluginFlyvemdmFleet::class,
+         'items_id_applied'            => $fleet->getID(),
+         'itemtype'                    => '',
+         'items_id'                    => '',
+      ]);
+      return [$policy, $task, $fleet];
```

Comment:
I would prefer a method to create a fleet (already exists), and an other to create a  task, given a fleet or an agent  as parameter.